projects/hello_adc/src/hello_adc.c

```diff
@@ -0,0 +1,39 @@
+#include "adc.h"
+#include "delay.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+
+#define BUTTON_PIN 178
+#define READ_FROM_PIN 166
+
+GpioItCallback prv_log_pin_data(GpioAddress *button_addr, void *context);
```

Comment:
This declares a function that *returns* a `GpioItCallback`, not one that *is* a `GpioItCallback`! Any function that matches the `GpioItCallback` can be used as one, you don't have to declare it implicitly; `GpioItCallback` is just a typedef for `void (*)(GpioAddress *, void *)` which is C-speak for "a pointer to a function that returns void and takes a GpioAddress pointer and a void pointer". So this needs to return `void` and not `GpioItCallback`.

Also, it should be `static` since it's a private function - that makes sure it's not visible to other files.